### PR TITLE
Make XML elements via server DO NOT MERGE

### PR DESCRIPTION
**Pull request by zerox1212**
for review

### CODE REVIEW

**Comment by zerox1212** [comment]
@oroulet Can you check ```node_to_xml``` and tell if that is what you had in mind?

**Comment by oroulet** [comment]
Looks OK for me but I am not an expert in xml generation.
The only thing is that maybe you should put together all queries:
browsename=node.get_browse_name()
localilzed_name = node.get_localizedxxxx

Later we may need to optimize the calls and it will be easier if they are grouped.

BUT I think we can have the xml exporter in the python-opcua repository, it might be used for other things than modeller There is already a file called xmlimporter.py. make a new one called xmlexporter.py

File: modeler/server-xml.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+def node_to_xml(etree, node):
+    if node.get_node_class() is ua.NodeClass.Variable:
+        var_el = ET.SubElement(etree.getroot(),
+                                'UAVariable',
+                                BrowseName=make_xml_browsename(node),
+                                DataType=o_ids.ObjectIdNames[node.get_data_type().Identifier],
+                                NodeId=make_xml_nodeid(node),
+                                ParentNodeId=get_parent(node),
+                                AccessLevel='3',  # TODO How to get this attribute as a number and not as AccessLevel.X enum
```
Comment (oroulet) [inline modeler/server-xml.py]: enum.val is the int. enum.name is the name :-)

File: modeler/server-xml.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                                UserAccessLevel='3')  # TODO How to get this attribute as a number and not as AccessLevel.X enum
+
+        disp_el = ET.SubElement(var_el, 'DisplayName',)
+        disp_el.text = node.get_display_name().Text.decode(encoding='UTF8')
+
+        refs_el = ET.SubElement(var_el, 'References')
+
+        refx_el = ET.SubElement(refs_el, 'Reference', ReferenceType='x')  # TODO find reusable way to get all refs
```
Comment (oroulet) [inline modeler/server-xml.py]: node.get_references(xxx)

File: modeler/server-xml.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+from collections import OrderedDict
+sys.path.insert(0, "..")
+import time
+
+from opcua import ua, Server
+from opcua.ua import object_ids as o_ids
+
+
+def node_to_xml(etree, node):
+    if node.get_node_class() is ua.NodeClass.Variable:
```
Comment (oroulet) [inline modeler/server-xml.py]: do the query once. not at every test. make av ariable node_class

**Comment by zerox1212** [comment]
Thanks for the review, I will start on an a module for python-opcua.

**Comment by oroulet** [comment]
closing this one. reopen it aginst python-opcua

**Pull request closed without merging**